packages/experimental/Menu/SPEC.md

```diff
@@ -0,0 +1,345 @@
+
+- `root` - The outer container representing the `MenuItem` itself that wraps everything passed via the `children` prop.
+- `content` - If specified, renders the `content` prop as text.
+- `checkmark` - If specified, renders space such that the `content` aligns with other items which have checkmarks. Only used when the `Menu` has `hasCheckmarks`.
+- `submenuIndicator` - If specified, renders an SVG which indicates that the `MenuItem` is a trigger for a nested `Menu`.
+
+### MenuItemCheckbox/Radio
+
+Variants of `MenuItem` that allows a single or multiple selection state based on the value that it represents. These `MenuItems` do not support submenus. `MenuItemCheckbox` and `MenuItemRadio` share the same types for props, tokens, and slots.
+
+#### MenuItemCheckbox/Radio Props
+
+```ts
+export interface MenuItemCheckboxProps extends Omit<IWithPressableOptions<ViewProps>, 'onPress'> {
+  content: string;
+
+  /**
+   * Applies disabled styles to menu item but remains focusable
+   */
+  disabled?: boolean;
+
+  /**
+   * A RefObject to access the IButton interface. Use this to access the public methods and properties of the component.
+   */
+  componentRef?: React.RefObject<IFocusable>;
+
+  /**
+   * Identifier for the control
+   */
+  name: string;
+}
+```
+
+#### MenuItemCheckbox/Radio Tokens
+
+```ts
+export interface MenuItemCheckboxTokens extends LayoutTokens, FontTokens, IBorderTokens, IColorTokens {
+  checkmarkColor?: ColorValue;
+  checkmarkPadding?: number;
+  checkmarkSize?: number;
+  checkmarkVisibility?: number;
+  gap?: number;
+
+  checked?: MenuItemCheckboxTokens;
+  disabled?: MenuItemCheckboxTokens;
+  focused?: MenuItemCheckboxTokens;
+  hovered?: MenuItemCheckboxTokens;
+  pressed?: MenuItemCheckboxTokens;
+}
+```
+
+#### MenuItemCheckbox/Radio Slots
+
+- `root` - The outer container representing the `MenuItem` itself that wraps everything passed via the `children` prop.
+- `content` - If specified, renders the `content` prop as text.
+- `checkmark` - If specified, renders an SVG which indicates that the `MenuItem` is selected.
+
+## Behaviors
+
+### Mouse interactions
+
+- Clicking on a menu's trigger will open the menu
+- Clicking outside of a menu's trigger and popover will light dismiss the menu
+- Clicking on a `MenuItem` which is not configured to track selection will invoke the `MenuItem` and close the menu by default
+- Clicking on a `MenuItem` variant which is configured to track selection will toggle its selection state in the configured way and not close the menu.
+- If the menu is configured to open on hover, it will only close once the mouse leaves both the trigger and the popover. There is a delay for this behavior.
+- Submenus are configured to open on hover by default. If the mouse stops hovering over the submenu, only the submenu will close.
+
+### Keyboard interactions
+
+- When Context menu is invoked via keyboard, open the menu and place focus on the first focusable element by default.
```

Comment:
Context menu has a slightly different meaning (I think there's a context menu key on keyboards?). Did you just mean menu? I don't think this really serves like a context menu